### PR TITLE
[OptApp] Fix for Project introduction

### PR DESCRIPTION
**Pull request by sunethwarna**
**📝 Description**
This PR enables `IndependentExecutionPolicy` to use the novel `Project` based `MultistageOrchestrator` for multi stages. This also enables other execution policies and all the `OptApp` related components to be loaded from sub paths of each application.

**🆕 Changelog**
- Extend `OptimizationFactory` for sub module load.
- Extend `IndependentExecutionPolicy` to support `MultistageOrchestrator`
- Update other execution policies
- Updated the test.

### CODE REVIEW

**Comment by matekelemen** [comment]
um
what's up with the CI?

**Comment by sunethwarna** [comment]
You mean why CI is not running? This is proposed to be merged into another branch, not the master :)

**Comment by matekelemen** [comment]
ah, right :facepalm:

File: applications/OptimizationApplication/python_scripts/utilities/helper_utilities.py [hunk 1]
```diff
@@ -8,26 +8,35 @@
 from KratosMultiphysics.kratos_utilities import GetKratosMultiphysicsPath
 from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import ContainerExpressionTypes
 
-def GetClassModuleFromKratos(class_name: str) -> str:
-    snake_case_class_name = Kratos.StringUtilities.ConvertCamelCaseToSnakeCase(class_name)
+def GetClassModuleFromKratos(full_class_name: str) -> str:
+    sub_module_paths = full_class_name.split(".")
+
+    if len(sub_module_paths) == 0:
+        raise RuntimeError("Empty class names are not allowed.")
```
Comment (philbucher) [inline applications/OptimizationApplication/python_scripts/utilities/helper_utilities.py]:
```suggestion
    if not sub_module_paths:
        raise RuntimeError("Empty class names are not allowed.")
```

**Review by rubenzorrilla — APPROVED** [review]
Very good. Thanks for the quick fix.

**Pull request merged**